public/app/plugins/datasource/tempo/datasource.ts

```diff
@@ -378,6 +399,60 @@ export class TempoDatasource extends DataSourceWithBackend<TempoQuery, TempoJson
     };
   }
 
+  handleMetricsSummary = (target: TempoQuery, query: string, options: DataQueryRequest<TempoQuery>) => {
+    const groupBy = target.groupBy
+      ?.filter((f) => f.tag)
+      .map((f) => {
+        if (f.scope === TraceqlSearchScope.Unscoped) {
+          return `.${f.tag}`;
+        }
+        return f.scope !== TraceqlSearchScope.Intrinsic ? `${f.scope}.${f.tag}` : f.tag;
+      })
+      .join(', ');
+
+    return this._request('/api/metrics/summary', {
+      q: query,
+      groupBy,
+      start: options.range.from.unix(),
+      end: options.range.to.unix(),
+    }).pipe(
+      startWith({
+        data: [],
+        state: LoadingState.Loading,
+      }),
+      map((response) => {
+        if (response.state !== LoadingState.Loading && !response.data.summaries) {
+          return {
+            error: {
+              message: getErrorMessage(
+                `No summary data for '${groupBy}'. Note: the metrics summary API only considers spans of kind = server. Please check if the attributes exist by running a TraceQL query like { attr_key = attr_value && kind = server }`
+              ),
+            },
+            data: createTableFrameFromMetricsSummaryQuery([], query, this.instanceSettings, LoadingState.Error),
+          };
+        }
+        return {
+          data: createTableFrameFromMetricsSummaryQuery(
+            response.data.summaries,
+            query,
+            this.instanceSettings,
+            LoadingState.Done
+          ),
+        };
+      }),
+      catchError((error) => {
+        return of({
+          error: { message: error.data.message },
+          data: createTableFrameFromMetricsSummaryQuery([], query, this.instanceSettings, LoadingState.Error),
```

Comment:
Do we need to create the data frame if we have error? Don't we handle that reasonably in Explore/Dashboards?

Reply:
We do so that we can update the state from loading to no data.

With empty data frame
![Screenshot 2023-08-23 at 08 53 38](https://github.com/grafana/grafana/assets/90795735/60d93346-09f7-45ff-b071-339687841740)

Without empty data frame
![Screenshot 2023-08-23 at 08 53 47](https://github.com/grafana/grafana/assets/90795735/e7481f64-e32b-4471-be09-3e80cf3e554b)